resources/service_sysv_init.rb

```diff
@@ -33,102 +34,103 @@
 
 property :setenv_template_source, String, default: 'setenv.erb'
 property :setenv_template_cookbook, String, default: 'tomcat'
+property :service_template_local, [true, false], default: false
 
 action :start do
-  create_init
+  action_create
 
-  service "tomcat_#{new_resource.instance_name}" do
+  service new_resource.service_name do
     provider platform_sysv_init_class
     supports restart: true, status: true
     action :start
   end
 end
 
+action :create do
+  # suse is missing libraries we need for sys-v
+  package 'perl-Getopt-Long-Descriptive' if platform_family?('suse')
+
+  # fedora can be missing chkconfig
+  package 'chkconfig' if node.platform?('fedora')
+
+  # define the lock dir for RHEL vs. debian
+  platform_lock_dir = value_for_platform_family(
+    %w(rhel fedora suse) => '/var/lock/subsys',
+    'debian' => '/var/lock',
+    'default' => '/var/lock'
+  )
+
+  # the init script will not run without redhat-lsb packages
+  if platform_family?('rhel', 'fedora')
+    if node['platform_version'].to_i < 6.0
```

Comment:
At this point, I think it's safe for us to go ahead and remove sysv init from any cookbook. EL6 is EOL at the end of November which is the last supported non-systemd platform.